apps/managedidentity/managedidentity.go

```diff
@@ -178,8 +207,47 @@ func createIMDSAuthRequest(ctx context.Context, id ID, resource string, claims s
 	return req, nil
 }
 
+// Contains checks if the element is present in the list.
+func contains[T comparable](list []T, element T) bool {
+	for _, v := range list {
+		if v == element {
+			return true
+		}
+	}
+	return false
+}
+
+// retry performs an HTTP request with retries based on the provided options.
+func retry(maxRetries int, c ops.HTTPClient, req *http.Request) (*http.Response, error) {
+	var resp *http.Response
+	var err error
+	for attempt := 0; attempt < maxRetries; attempt++ {
+		tryCtx, tryCancel := context.WithTimeout(req.Context(), time.Second*15)
+		defer tryCancel()
+		cloneReq := req.Clone(tryCtx)
+		resp, err = c.Do(cloneReq)
+		if err == nil && !contains(retryStatusCodes, resp.StatusCode) {
```

Comment:
This should consider the Source. `retryStatusCodes` follows IMDS docs, and that guidance doesn't make sense for other APIs. For example, in general you don't want to retry 404. Azure SDK retries [these codes](https://github.com/Azure/azure-sdk-for-go/blob/5635b52606e8f156575c11317b87a2d51013b710/sdk/azcore/runtime/policy_retry.go#L51-L56) by default and special-cases IMDS

Reply:
CC @neha-bhargava as this is probably a small bug in all other MSALs (which should not affect Azure SDK)

Reply:
After discussions with @bgavrilMS, we have decided to maintain consistency with other MSAL implementations. We will create a bug report to address this issue and ensure that the necessary updates are applied across all relevant areas.